### PR TITLE
tech(ApiAsyncRefactor) Adding withAsync to other endpoints and cleaning up the flow of code in the API

### PR DESCRIPTION
**Pull request by dejanvasic85**

### CODE REVIEW

File: server/routes/clients.js [hunk 1]
```diff
@@ -2,31 +2,32 @@ const express = require('express');
...
+
+router.get('/:id', withAsync(async (req, res) => {
   const agentId = req.agent.id;
   const id = req.params.id;
 
   logger.info(`Fetch by client id ${id}, agentId: ${agentId}`);
   const client = await Client.findOne({ where: { id, agentId } });
   if (!client) {
-    res.json({ error: 'unknown client id' }).status(404);
+    res.status(404).json({ error: 'unknown client id' });
```
Comment (dejanvasic85) [inline server/routes/clients.js:30]: This always trips me up but I think that status has to go before `json()` or `send()`

File: server/routes/prospects.js [hunk 2]
```diff
@@ -2,59 +2,77 @@ const express = require('express');
...
-    res.status(201).json(p);
-  }).catch(err => {
-    res.status(500).json({ err });
-  });
-});
+  const agent = await Agent.findOne({ where: { id: newProspect.agentId } });
+  const client = await Client.findOne({ where: { id: newProspect.clientId } });
+  const prospect = await Prospect.create(newProspect);
+  // Should we even await this?
+  await emailer.sendNewProspectEmail(newProspect, client, agent);
```
Comment (dejanvasic85) [inline server/routes/prospects.js:72]: This sends out 3 different emails! Should we be waiting for it? What happens on failure? What if only 1 email is sent and the others are not?

Comment (FastNinja) [reply server/routes/prospects.js:72]: well... what is likelihood of that happened? and if happened what is the damage?

based on ^^ . - I think it is acceptable risk.. Sendgrid is fast.. if sending emails fails - you want user to know -right?

**Review by FastNinja — APPROVED** [review]

File: server/routes/prospects.js (hunk 2 above)
Comment (dejanvasic85) [reply server/routes/prospects.js:72]: Yeah, i found it to be a little bit slow, but it's ok. We'll try it out.

**Pull request merged**